docs/ape17.rst

```diff
@@ -0,0 +1,537 @@
+
+   git commit -m "Remove astropy-helpers submodule"
+
+Step 2: Update/create ``setup.cfg``
+-----------------------------------
+
+The next step is to update make sure that you have a ``setup.cfg`` file
+that contains meta-data about the package. If you already have this
+file, you will likely need to update it, and if you don’t already have
+this file, you will need to create it.
+
+This file should contain at least the following entries::
+
+   [metadata]
+   name = my-package
+   author = ...
+   author_email = ...
+   license = ...
+   license_file = LICENSE.rst
+   url = ...
+   description = ...
+   long_description = file: README.rst
+
+   [options]
+   zip_safe = False
+   packages = find:
+   install_requires =
+       numpy
+       astropy
+       ...
+   python_requires = >=3.6
+
+Replace the ``...`` with the information for your package. Make sure
+that ``license_file`` and ``long_description`` have the right filename,
+and specify your required dependencies one per line in the
+``install_requires`` section. Make sure the ``python_requires`` line is
+set to indicate the correct minimum Python version for your package.
+
+If you already had a file, make sure you remove the following entries
+(if present):
+
+-  ``package_name``
+-  ``version``
+-  ``setup_requires``
+-  ``tests_require``
+-  ``[ah_bootstrap]`` and all entries in it
+
+Step 3 - Define optional, test, and docs dependencies
+-----------------------------------------------------
+
+Next up, add a new section to the ``setup.cfg`` file (or modify, if it
+already exists), to specify optional dependencies as well as
+dependencies required to run tests and build the documentation, for
+example::
+
+   [options.extras_require]
+   all =
+       scipy
+       matplotlib
+   test =
+       pytest-astropy
+   docs =
+       sphinx-astropy
+
+If you don’t need any optional dependencies, remove the ``all`` section.
+You will likely need to have at least ``pytest-astropy`` in the ``test``
+section and ``sphinx-astropy`` in ``docs``.
+
+Step 4 - Define package data
+----------------------------
+
+If your package includes non-Python data files, you will need to update
+how you declare which data files to include. If you have been using the
+Astropy package template, it is likely that you have functions called
+``get_package_data`` defined inside ``setup_package.py`` files. Remove
+these functions, and instead define the package data using a
+``[options.package_data]`` section inside your ``setup.cfg`` file,
+e.g.::
+
+   [options.package_data]
+   * = *.fits, *.csv
+   my_package.tests = data/*
+
+In the above example, all ``.fits`` and ``.csv`` in the package will be
+included as well as all files inside ``my_package/tests/data``.
+
+Step 5 - Update your ``setup.py`` file
+--------------------------------------
+
+Copy the ``setup.py`` file you generated in :ref:`Step 0 <step0>` and replace your existing one
+- it should be good to go as-is without any further customizations.
```

Comment:
The master branch of the repo is the rendered version with defaults, so we can point to that.

Reply:
@mwcraig - the issue is that then the rendered version does need a customization. Even if you already have a package going the cookiecutter step 0 does make things a lot easier. Same goes for the tox file where the package name has to be customized. Maybe I should clarify this in Step 0?

Reply:
The other issue is that the setup.py file is different whether or not there are compiled extensions. So it really is easier to just run cookiecutter yourself in Step 0 and then copy the file over.

Reply:
I've re-worded Step 0 to make it clearer this is not an optional step - see changes in e1658a33dc75102ebcba1078fd1c622409949841. @mwcraig - does that make it clearer?

Just to be clear, I could link to the rendered versions in master, I'm just worried it will add a whole bunch of text to explain the changes that have to be made to them.

Reply:
> I've re-worded Step 0 to make it clearer this is not an optional step - see changes in e1658a3. @mwcraig - does that make it clearer?

Yep, this looks much clearer.